### DISCUSSION TITLE
Deprecation/versioning documentation

### DISCUSSION THREAD

**Discussion by ananthsub** [Lightning-AI/pytorch-lightning#8879] [General]
How should we communicate Lightning's versioning & deprecation on the docs? The current procedure of mark as deprecated in version N and deprecate in minor version N+2 isn't mentioned anywhere, but I think this would be valuable to communicate that:
- Lightning has a formal procedure through which we go through API deprecations
- Lightning takes backwards compatibility very seriously
- Lightning as a project is perpetually evolving and improving

I think it'd also be useful to publish:
- Schedule for patch versions, and what they contain (bugfixes)
- Schedule for minor versions, and what they contain (e.g. new features, bugfixes, deprecations)
- What would qualify/necessitate a new major version

@edenafek

**Comment by justusschock** [comment]
cc @edenlightning

**Comment by awaelchli** [comment]
I opened #9025 to kickstart this.
